### ISSUE TITLE
Bug when auto closing menu

### ISSUE THREAD

**Issue by alnitak** [alnitak/flutter_star_menu#9]
ref to the flow chart package issue [Text Editor Bug/Crash #2 ](https://github.com/alnitak/flutter_flow_chart/issues/2)

**Comment by jdar** [comment]
Is there a suggested workaround, until this is resolved? 

Or perhaps you can describe the design path to a resolution so that I can attempt a PR?
(Also, I am very confused by latest commit (49e6b5630aeaa lazy items) ... perhaps you can integrate that into the example/ so that I can understand better)

**Comment by alnitak** [comment]
The latest commit added all the parameters that StarMenu widget accept, available also to the addStarMenu mixin. So, since only ```items``` or ```lazyItems``` parameter can be passed, I needed to make all those optional. The example is also updated to fix that (everywhere ```addStarMenu``` is used).

Back to the issue, the problem comes up when a TextField is used as a menu item: when you touch the TextField the virtual keyboard pops up resizing the StarMenu overlay, hence  it triggers a refresh [here](https://github.com/alnitak/flutter_star_menu/blob/59a3cb3a7f11e2ff4c618f6f89513e49a85d2d31/lib/src/star_menu.dart#L225).
The refresh implies a menu close and reopen disposing the TextField and, because of this, the virtual keyboard hides and the user cannot input the text.
I tried to look at a fix or a workaround, but didn't find a solution yet
